Signal/src/ViewControllers/AddToGroupViewController.m

```diff
@@ -0,0 +1,127 @@
+}
+
+- (NSString *)phoneNumberSectionTitle
+{
+    return NSLocalizedString(@"ADD_GROUP_MEMBER_VIEW_PHONE_NUMBER_TITLE",
+        @"Title for the 'add by phone number' section of the 'add group member' view.");
+}
+
+- (NSString *)phoneNumberButtonText
+{
+    return NSLocalizedString(@"ADD_GROUP_MEMBER_VIEW_BUTTON",
+        @"A label for the 'add by phone number' button in the 'add group member' view");
+}
+
+- (NSString *)contactsSectionTitle
+{
+    return NSLocalizedString(
+        @"ADD_GROUP_MEMBER_VIEW_CONTACT_TITLE", @"Title for the 'add contact' section of the 'add group member' view.");
+}
+
+- (void)phoneNumberWasSelected:(NSString *)phoneNumber
+{
+    OWSAssert(phoneNumber.length > 0);
+
+    __weak AddToGroupViewController *weakSelf = self;
+    ContactsViewHelper *helper = self.contactsViewHelper;
+    if ([helper isRecipientIdBlocked:phoneNumber]) {
+        [BlockListUIUtils showUnblockPhoneNumberActionSheet:phoneNumber
+                                         fromViewController:self
+                                            blockingManager:helper.blockingManager
+                                            contactsManager:helper.contactsManager
+                                            completionBlock:^(BOOL isBlocked) {
+                                                if (isBlocked) {
+                                                    [weakSelf addToGroup:phoneNumber];
+                                                }
+                                            }];
+    } else {
+        [self addToGroup:phoneNumber];
+    }
+}
+
+- (void)signalAccountWasSelected:(SignalAccount *)signalAccount
+{
+    OWSAssert(signalAccount);
+
+    __weak AddToGroupViewController *weakSelf = self;
+    ContactsViewHelper *helper = self.contactsViewHelper;
+    if ([helper isRecipientIdBlocked:signalAccount.recipientId]) {
+        [BlockListUIUtils showUnblockSignalAccountActionSheet:signalAccount
+                                           fromViewController:self
+                                              blockingManager:helper.blockingManager
+                                              contactsManager:helper.contactsManager
+                                              completionBlock:^(BOOL isBlocked) {
+                                                  if (isBlocked) {
+                                                      [weakSelf addToGroup:signalAccount.recipientId];
+                                                  }
+                                              }];
+    } else {
+        [self addToGroup:signalAccount.recipientId];
+    }
+}
+
+- (void)addToGroup:(NSString *)recipientId
+{
+    OWSAssert(recipientId.length > 0);
+
+    [self.addToGroupDelegate recipientIdWasAdded:recipientId];
+    [self.navigationController popViewControllerAnimated:YES];
+}
+
+- (BOOL)shouldHideLocalNumber
+{
+    return YES;
+}
+
+- (BOOL)shouldHideContacts
+{
+    return self.hideContacts;
```

Comment:
The idea was this:

* In "update group", you're making a small narrow edit to a group, usually adding one person.  So we depend entirely on the "add to group" view which will let you add by phone number or pick from a list of your contacts.
* In "new group", you're often going to want to pick many users so we include the contacts list right in the "new group" view.  We don't want to show the contacts list again in the "add by phone number" view.